### PR TITLE
[Merged by Bors] - Revert ndk-glue to 0.5 to synchronize with winit

### LINKED ISSUE #4529: Android: Panic unwrapping None value while getting Native Activity. Native Activity is never not None.
## Bevy version

Bevy 0.8.0 dev, cloned today.

## Operating system & version
Windows 10, targeting android

## What you did

`cargo apk run --example android`

I get this when ran without any modifications
```
rust.example.android I/RustStdoutStderr: thread '<unnamed>' panicked at 'called 'Option::unwrap()' on a 'None' value', [...]\ndk-glue-0.6.1\src\lib.rs:60:39
```

which is 
```
    unsafe { NATIVE_ACTIVITY.as_ref().unwrap() }
```

When I modify the unsafe call to 
```
    //unsafe { NATIVE_ACTIVITY.as_ref().unwrap() }
    println!("Getting activity");
    unsafe {
      loop {
        let a = NATIVE_ACTIVITY.as_ref();
        match a {
          Some(_) => {
            println!("Found activity");
            ()
          },
          None => (),
        }
      }
    }
```

I get the print message that it's getting activity, but it never gets a non-none value. Might be related to https://github.com/rust-windowing/android-ndk-rs/issues/211

Full log while waiting for activity
```
adb logcat --pid $(adb shell pidof -s rust.example.android)
--------- beginning of main
04-18 14:19:52.353 10452 10452 I example.androi: Late-enabling -Xcheck:jni
04-18 14:19:52.410 10452 10452 I example.androi: Unquickening 20 vdex files!
04-18 14:19:52.602 10452 10452 W System  : ClassLoader referenced unknown path:
04-18 14:19:52.669 10452 10452 D NetworkSecurityConfig: No Network Security Config specified, using platform default
04-18 14:19:52.672 10452 10452 D NetworkSecurityConfig: No Network Security Config specified, using platform default
04-18 14:19:52.922 10452 10452 W linker  : Warning: "/data/app/~~lYnEs5tBaOKmqYDPgA8DjQ==/rust.example.android-Aaqia3cBRGEk1FzLFjB99g==/lib/arm64/libc++_shared.so" unused DT entry: unknown processor-specific (type 0x70000001 arg 0x0) (ignoring)
04-18 14:19:52.968 10452 10507 I RustStdoutStderr: Waiting for NativeScreen
04-18 14:19:57.233 10452 10507 I RustStdoutStderr: NativeScreen Found:Some(NativeWindow { ptr: 0xb4000077254f24c0 })
04-18 14:19:57.510 10452 10508 D vulkan  : searching for layers in '/data/app/~~lYnEs5tBaOKmqYDPgA8DjQ==/rust.example.android-Aaqia3cBRGEk1FzLFjB99g==/lib/arm64'
04-18 14:19:57.511 10452 10508 D vulkan  : searching for layers in '/data/app/~~lYnEs5tBaOKmqYDPgA8DjQ==/rust.example.android-Aaqia3cBRGEk1FzLFjB99g==/base.apk!/lib/arm64-v8a'
04-18 14:19:57.540 10452 10508 I AdrenoVK-0: ===== BEGIN DUMP OF OVERRIDDEN SETTINGS =====
04-18 14:19:57.540 10452 10508 I AdrenoVK-0: ===== END DUMP OF OVERRIDDEN SETTINGS =====
04-18 14:19:57.541 10452 10508 I AdrenoVK-0: QUALCOMM build          : d4cfdf3, Ic907de5ed0
04-18 14:19:57.541 10452 10508 I AdrenoVK-0: Build Date              : 09/25/20
04-18 14:19:57.541 10452 10508 I AdrenoVK-0: Shader Compiler Version : EV031.32.02.01
04-18 14:19:57.541 10452 10508 I AdrenoVK-0: Local Branch            : mybrancheaff3d46-e51f-f136-8926-1458d0b52af0
04-18 14:19:57.541 10452 10508 I AdrenoVK-0: Remote Branch           : quic/gfx-adreno.lnx.1.0.r116-rel
04-18 14:19:57.541 10452 10508 I AdrenoVK-0: Remote Branch           : NONE
04-18 14:19:57.541 10452 10508 I AdrenoVK-0: Reconstruct Branch      : NOTHING
04-18 14:19:57.541 10452 10508 I AdrenoVK-0: Build Config            : S P 10.0.7 AArch64
04-18 14:19:57.541 10452 10508 I AdrenoVK-0: Driver Path             : /vendor/lib64/hw/vulkan.adreno.so
04-18 14:19:57.572 10452 10508 I event crates\bevy_render\src\renderer\mod.rs:98:  AdapterInfo { name: "Adreno (TM) 610", vendor: 20803, device: 100728832, device_type: IntegratedGpu, backend: Vulkan }
04-18 14:19:57.710 10452 10507 I RustStdoutStderr: Getting activity
04-18 14:21:45.869 10452 10483 I example.androi: Thread[3,tid=10483,WaitingInMainSignalCatcherLoop,Thread*=0xb4000077154d9010,peer=0x12d40240,"Signal Catcher"]: reacting to signal 3
04-18 14:21:45.870 10452 10483 I example.androi:
04-18 14:21:46.636 10452 10483 I example.androi: Wrote stack traces to tombstoned
```

### LINKED ISSUE #4774: Revert ndk-glue version or update to new winit release
#3624 updated our ndk-glue version, but this will break Android audio further, as the version must match the version used by `winit`.

This is [updated on `winit` main](https://github.com/rust-windowing/winit/blob/master/Cargo.toml), so we can either tackle this by reverting the change, or migrating versions.

Because Android is already broken in more serious ways, I'm not immediately concerned. However I'm adding this to the milestone so we do at least one of those things before the next release.

### PR DESCRIPTION
**Pull request by alice-i-cecile**
# Objective

- Upgrading ndk-glue (our Android interop layer) desynchronized us from winit
- This further broke Android builds, see #4905 (oops...)
- Reverting to 0.5 should help with this, until the new `winit` version releases
- Fixes #4774 and closes #4529

### CODE REVIEW

**Comment by cart** [comment]
> While I'm there, synchronize features relied upon in our dependencies to avoid compiling in two slightly different versions of the crate.

Aren't cargo features additive for a build? I thought only including the exact features you need for a specific crate was the "preferred approach".

**Comment by mockersf** [comment]
> > While I'm there, synchronize features relied upon in our dependencies to avoid compiling in two slightly different versions of the crate.
> 
> Aren't cargo features additive for a build? I thought only including the exact features you need for a specific crate was the "preferred approach".

Yup, no need to copy the features.

File: crates/bevy_asset/Cargo.toml [hunk 1]
```diff
@@ -40,7 +40,7 @@ wasm-bindgen-futures = "0.4"
 js-sys = "0.3"
 
 [target.'cfg(target_os = "android")'.dependencies]
-ndk-glue = { version = "0.6" }
+ndk-glue = { version = "0.5"}
```
Comment (alice-i-cecile) [inline crates/bevy_asset/Cargo.toml:43]:
```suggestion
ndk-glue = { version = "0.5" }
```

File: crates/bevy_internal/Cargo.toml [hunk 2]
```diff
@@ -98,4 +98,4 @@ bevy_winit = { path = "../bevy_winit", optional = true, version = "0.8.0-dev" }
 bevy_gilrs = { path = "../bevy_gilrs", optional = true, version = "0.8.0-dev" }
 
 [target.'cfg(target_os = "android")'.dependencies]
-ndk-glue = {version = "0.6", features = ["logger"]}
+ndk-glue = {version = "0.5", features = ["logger"]}
```
Comment (MarijnS95) [inline crates/bevy_internal/Cargo.toml:101]: Same comment regarding spaces here?

```suggestion
ndk-glue = { version = "0.5", features = ["logger"] }
```

**Comment by MarijnS95** [comment]
Perhaps you want to leave a comment about this in Cargo.toml too, and refer to the compatibility table winits' README.

File: crates/bevy_internal/Cargo.toml (hunk 2 above)
Comment (alice-i-cecile) [reply crates/bevy_internal/Cargo.toml:101]: Eh I don't mind it as much here; the other one was imbalanced.

Comment (alice-i-cecile) [inline crates/bevy_internal/Cargo.toml:101]:
```suggestion
// This version *must* be the same as the version used by winit,
// or Android will break: https://github.com/rust-windowing/winit#android
ndk-glue = {version = "0.5", features = ["logger"]}
```

**Review by mockersf — APPROVED** [review]

File: crates/bevy_internal/Cargo.toml (hunk 2 above)
Comment (MarijnS95) [reply crates/bevy_internal/Cargo.toml:103]: It's very inconsistent with the rest of the file :see_no_evil:, but you do you...

**Pull request closed without merging**